python/tank_vendor/shotgun_deploy/configuration.py

```diff
@@ -216,21 +204,43 @@ def update_configuration(self):
 
         # @todo - prime caches (yaml, path cache)
 
-        # create tank executable for this config - based on the default desktop
-        # python interpreter settings
+        # make sure tank command and interpreter files are up to date
         self._create_tank_command()
 
         if self._pipeline_config_id:
-            # create pipeline configuration entry in Shotgun
-            log.debug("Updating pipeline configuration %s with new paths..." % self._pipeline_config_id)
-            self._sg_connection.update(
+            # make sure there is a pipeline config entry in Shotgun
+            # and that this is up to date. We may not have permission
+            # to write to this configuration, so take a conservative
+            # approach where we first check if the record exists and is
+            # up to date and only if it differs we attempt to update it.
+            log.debug(
+                "Checking that shotgun pipeline config entry "
+                "id %s exists and is up to date..." % self._pipeline_config_id
+            )
+
+            pc_data = self._sg_connection.find_one(
                 constants.PIPELINE_CONFIGURATION_ENTITY_TYPE,
-                self._pipeline_config_id,
-                {"linux_path": self._paths["linux2"],
-                 "windows_path": self._paths["win32"],
-                 "mac_path": self._paths["darwin"]}
+                [["id", "is", self._pipeline_config_id]],
+                ["linux_path", "windows_path", "mac_path"]
             )
 
+            log.debug("Shotgun data returned: %s" % pc_data)
+
+            if pc_data is None or \
+                pc_data.get("linux_path") != self._paths["linux2"] or \
+                pc_data.get("windows_path") != self._paths["win32"] or \
+                pc_data.get("mac_path") != self._paths["darwin"]:
+
+                log.debug("Attempting to updating pipeline configuration with new paths...")
+
+                self._sg_connection.update(
+                    constants.PIPELINE_CONFIGURATION_ENTITY_TYPE,
+                    self._pipeline_config_id,
+                    {"linux_path": self._paths["linux2"],
+                     "windows_path": self._paths["win32"],
```

Comment:
It could also be argued that since we are going to compare paths from the pipeline configuration with self._paths, we might as well use the same keys?